src/NUnitTestAdapter/NUnit3TestExecutor.cs

```diff
@@ -135,6 +147,12 @@ public void RunTests(IEnumerable<string> sources, IRunContext runContext, IFrame
             Unload();
         }
 
+        private void SetRunTypeByStrings() =>
+            RunType = !Settings.DesignMode
+                ? Settings.DiscoveryMethod == DiscoveryMethod.Legacy ? RunType.CommandLineLegacy :
+                Settings.UseNUnitFilter ? RunType.CommandLineCurrentNUnit : RunType.CommandLineCurrentVsTest
+                : RunType.Ide;
```

Comment:
This is difficult to follow. I'd recommend nested, where there is a newline before every `?` and `:`:

```cs
            RunType = !Settings.DesignMode
                ? Settings.DiscoveryMethod == DiscoveryMethod.Legacy
                    ? RunType.CommandLineLegacy
                    : Settings.UseNUnitFilter
                        ? RunType.CommandLineCurrentNUnit
                        : RunType.CommandLineCurrentVsTest
                : RunType.Ide;
```

Or linear (when the structure permits it), where there is a newline before every condition and before the final 'else' value:

```cs
            RunType =
                Settings.DesignMode ? RunType.Ide :
                Settings.DiscoveryMethod == DiscoveryMethod.Legacy ? RunType.CommandLineLegacy :
                Settings.UseNUnitFilter ? RunType.CommandLineCurrentNUnit :
                RunType.CommandLineCurrentVsTest;
```

Reply:
Done

Reply:
C# 8 option just for comparison:

```cs
            RunType = Settings switch
            {
                { DesignMode: true} => RunType.Ide,
                { DiscoveryMethod: DiscoveryMethod.Legacy } => RunType.CommandLineLegacy,
                { UseNUnitFilter: true} => RunType.CommandLineCurrentNUnit,
                _ => RunType.CommandLineCurrentVsTest,
            };
```

Reply:
This last one is very good.  I'll change to that one on the next PR.  Thanks @jnm2